src/hooks/useTokenIdentityRegistryAgents.ts

```diff
@@ -34,6 +37,33 @@ export function useTokenIdentityRegistryAgents(registry?: EvmAddress) {
     }
   }, [registry]);
 
+  const filteredNotAgentsYetQueryResult = useQueries({
```

Comment:
Not quite sure why do we need 2 almost identical calls for `filteredAgentsQueryResult` and `filteredNotAgentsYetQueryResult`?

Can we keep only one, the one with already filtered duplicate agents? Unless we have some use case for not filtered ones?

Also I'd suggest to move out this `useQueries` call as separate hook and use in current hook if needed afterwards.